userspace/libsinsp/tracer_emitter.cpp

```diff
@@ -177,3 +182,13 @@ void tracer_emitter::write_tracer(const bool enter)
 		m_exit_written = true;
 	}
 }
+
+uint64_t tracer_emitter::elapsed_time() const
+{
+	auto elapsed = sinsp_utils::get_current_time_ns() - m_start_ns;
```

Comment:
Ditto on externally provided time, although you may only be able to specify it when explicitly calling stop().